pkg/intermediate/aggregate.go

```diff
@@ -270,21 +273,35 @@ func addOriginalExporterInfo(message *entities.Message) error {
 	set := message.Set
 	records := set.GetRecords()
 	for _, record := range records {
-		var originalExporterIPv4Address, originalObservationDomainId *entities.InfoElementWithValue
-
-		// Add originalExporterIPv4Address
-		ie, err := registry.GetInfoElement("originalExporterIPv4Address", registry.IANAEnterpriseID)
-		if err != nil {
-			return fmt.Errorf("IANA Registry is not loaded correctly with originalExporterIPv4Address.")
+		var originalExporterAddress, originalObservationDomainId *entities.InfoElementWithValue
+		var ie *entities.InfoElement
+		var address net.IP
+		var err error
+		// Add original exporter address
+		if net.ParseIP(message.ExportAddress).To4() != nil { // IPv4
+			ie, err = registry.GetInfoElement("originalExporterIPv4Address", registry.IANAEnterpriseID)
+			if err != nil {
+				return fmt.Errorf("IANA Registry is not loaded correctly with originalExporterIPv4Address.")
+			}
+			address = net.ParseIP(message.ExportAddress).To4()
```

Comment:
You may rearrange the code so that net.ParseIP() can be executed only once (i.e. lines 281 and 286, also lines 287 and 292).